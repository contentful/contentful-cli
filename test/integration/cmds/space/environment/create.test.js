const nixt = require('nixt')
const { join } = require('path')
const { initConfig, deleteSpaces, createSimpleSpace } = require('../../../util')

const bin = join(__dirname, './../../../../../', 'bin')

const app = () => {
  return nixt({ newlines: true })
    .cwd(bin)
    .base('./contentful.js ')
    .clone()
}

const org = process.env.CLI_E2E_ORG_ID
let space = null
const spacesToDelete = []

beforeAll(async () => {
  await initConfig()
  space = await createSimpleSpace(org, 'space-env-cr')
  spacesToDelete.push(space.sys.id)
})

afterAll(() => {
  return deleteSpaces(spacesToDelete)
})

test('should exit 1 when no args', done => {
  app()
    .run('space environment create')
    .code(1)
    .expect(result => {
      const resultText = result.stderr.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should print help message', done => {
  app()
    .run('space environment create --help')
    .code(0)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should create environment with id and name provided', done => {
  app()
    .run(
      `space environment create --space-id ${space.sys.id} --environment-id staging --name Staging`
    )
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot()
    })
    .code(0)
    .end(done)
})

test('should create environment using shortcuts args', done => {
  app()
    .run(
      `space environment create -s ${space.sys.id} -e shortcutenvironment -n shortcutEnvironment`
    )
    .stdout(
      /Successfully created environment shortcutEnvironment \(shortcutenvironment\)/
    )
    .code(0)
    .end(done)
})
