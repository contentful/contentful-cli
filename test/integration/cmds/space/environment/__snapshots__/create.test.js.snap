// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create environment with id and name provided 1`] = `"Successfully created environment Staging (staging)"`;

exports[`should exit 1 when no args: help data is incorrect 1`] = `
"Usage: contentful space environment create --name 'Your Environment Name'

Options:
  -h, --help                Show help                                  [boolean]
  --environment-id, -e      Id of the environment to create           [required]
  --name, -n                Name of the environment to create         [required]
  --source, --src           ID of the source environment to create the new
                            environment from                            [string]
  --space-id, -s            ID of the space that the environment will belong to
                                                                        [string]
  --await-processing, -w    Wait until the environment is processed and ready
                                                      [boolean] [default: false]
  --management-token, --mt  Contentful management API token             [string]

See more at:
https://github.com/contentful/contentful-cli/tree/master/docs/space/environment/
create
Copyright 2019 Contentful
Missing required arguments: environment-id, name"
`;

exports[`should print help message: help data is incorrect 1`] = `
"Usage: contentful space environment create --name 'Your Environment Name'

Options:
  -h, --help                Show help                                  [boolean]
  --environment-id, -e      Id of the environment to create           [required]
  --name, -n                Name of the environment to create         [required]
  --source, --src           ID of the source environment to create the new
                            environment from                            [string]
  --space-id, -s            ID of the space that the environment will belong to
                                                                        [string]
  --await-processing, -w    Wait until the environment is processed and ready
                                                      [boolean] [default: false]
  --management-token, --mt  Contentful management API token             [string]

See more at:
https://github.com/contentful/contentful-cli/tree/master/docs/space/environment/
create
Copyright 2019 Contentful"
`;
